iib/workers/tasks/build.py

```diff
@@ -337,6 +341,104 @@ def _get_resolved_image(pull_spec):
     return pull_spec_resolved
 
 
+@retry(attempts=2, wait_on=IIBError, logger=log)
+def _get_present_bundles(from_index, base_dir):
```

Comment:
Should we split this one big function into multiple helper functions? I feel this function is doing too many things at once.

Reply:
Done